akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -184,4 +183,9 @@ object Sink extends SinkApply {
   def actorSubscriber[T](props: Props): Sink[T, ActorRef] =
     new Sink(new ActorSubscriberSink(props, DefaultAttributes.actorSubscriberSink, shape("ActorSubscriberSink")))
 
+  def queue[T](bufferSize: Int, timeout: FiniteDuration = 5.seconds): Sink[T, () ⇒ Future[Option[T]]] = {
```

Comment:
I know that the raw function type is elegant ("() ⇒ Future[Option[T]]") but I would prefer a Trait instead with a named method, "pull()" for example.